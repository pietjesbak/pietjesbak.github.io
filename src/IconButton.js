import React, { Component } from 'react';
import Tooltip from "react-simple-tooltip";

class IconButton extends Component {
    constructor(props) {
        super(props);

        this.state = {
            windowWidth: window.innerWidth
        };
    }

    handleResize = (e) => {
        this.setState({
            windowWidth: window.innerWidth
        });
    }

    componentDidMount() {
        window.addEventListener('resize', this.handleResize)
    }

    componentWillUnmount() {
        window.removeEventListener('resize', this.handleResize)
    }

    render() {
        if (this.state.windowWidth <= 500) {
            return (
                <Tooltip content={this.props.text} placement={this.props.placement}>
                    <button className={"small-button " + this.props.subClass} onClick={this.props.action}>
                        <i className={"icon-" + this.props.icon}></i>
                    </button>
                </Tooltip>
            );
        } else {
            return (
                <button className={"button " + this.props.subClass} onClick={this.props.action}>
                    {this.props.text}
                </button>
            );
        }
    }
}

export default IconButton;
