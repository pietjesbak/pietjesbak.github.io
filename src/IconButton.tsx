import * as React from 'react';
import Tooltip from 'react-simple-tooltip';
import { ICONBUTTON_MIN_SCREEN_WIDTH } from './data/Constants';

export interface State {
    windowWidth: number;
}

export interface Props extends React.HTMLAttributes<IconButton> {
    text: string | JSX.Element;
    placement?: string;
    subClass?: string;
    icon: string;
    to?: string;

    action?: (e: React.SyntheticEvent) => void;
}

export default class IconButton extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);

        this.state = {
            windowWidth: window.innerWidth
        };
    }

    handleResize = () => {
        this.setState({
            windowWidth: window.innerWidth
        });
    }

    componentDidMount() {
        window.addEventListener('resize', this.handleResize)
    }

    componentWillUnmount() {
        window.removeEventListener('resize', this.handleResize)
    }

    render() {
        if (this.state.windowWidth <= ICONBUTTON_MIN_SCREEN_WIDTH) {
            return (
                <Tooltip content={this.props.text} placement={this.props.placement}>
                    <button className={"small-button " + this.props.subClass} onClick={this.props.action}>
                        <i className={"icon-" + this.props.icon} />
                    </button>
                </Tooltip>
            );
        } else {
            return (
                <button className={"button " + this.props.subClass} onClick={this.props.action}>
                    <i className={"icon-" + this.props.icon} /> {this.props.text}
                </button>
            );
        }
    }
}
