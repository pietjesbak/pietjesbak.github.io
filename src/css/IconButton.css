button, .button, .small-button, .button-holder > * {
    display: inline-block;
    text-align: center;
    text-decoration: none;
    border-radius: 2px;
    background-color: #FA6900;
    cursor: pointer;
    border: 0;
    min-width: 120px;
    color: #fff!important;
    font-size: 14px;
    box-shadow: 0 2px 5px 0 rgba(0, 0, 0, .26);
    padding: 10px 6px;
    box-sizing: content-box;
}

button:hover, .button:hover, .small-button:hover, .button.active, .small-button.active {
    position: relative;
    top: 2px;
    box-shadow: 0 -2px 5px 0 rgba(0, 0, 0, .26);
}

button[disabled], .button[disabled], .small-button[disabled] {
    background-color: #d1681c;
}

.button-holder > *:hover {
    background-color: #fff;
}

.small-button {
    min-width: 40px;
    width: 40px;
}

.button-wrapper {
    display: inline-block;
    position: relative;
}

.button-holder {
    position: absolute;
    z-index: 100;
    margin: 0!important;
}

.button-holder.left {
    left: 0;
}

.button-holder.right {
    right: 0;
}

.button-holder > * {
    display: block;
    background-color: rgb(250, 229, 215);
    color: #000!important;
    font-weight: 300;
    position: initial!important; /** These buttons do not get pressed in on hover. */
}
