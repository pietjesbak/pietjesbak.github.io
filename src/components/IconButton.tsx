import '../css/IconButton.css';

import classNames from 'classnames';
import * as React from 'react';
import Tooltip from 'react-simple-tooltip';
import { ICONBUTTON_MIN_SCREEN_WIDTH } from '../data/Constants';

export interface State {
    windowWidth: number;
    active: boolean;
}

export enum IconButtonBehavour {
    SMALL,
    BIG,
    AUTO
}

export interface Props extends React.HTMLAttributes<IconButton> {
    text: string | JSX.Element;
    placement?: string;
    subClass?: string;
    icon: string;
    to?: string;
    behaviour?: IconButtonBehavour

    action?: (e: React.SyntheticEvent) => void;
}

export default class IconButton extends React.Component<Props, State> {
    static defaultProps = {
        behaviour: IconButtonBehavour.AUTO
    }

    constructor(props: Props) {
        super(props);

        this.state = {
            windowWidth: window.innerWidth,
            active: false
        };
    }

    handleResize = () => {
        if (this.props.behaviour === IconButtonBehavour.AUTO) {
            this.setState({
                windowWidth: window.innerWidth
            });
        }
    }

    componentDidMount() {
        window.addEventListener('resize', this.handleResize);
    }

    componentWillUnmount() {
        window.removeEventListener('resize', this.handleResize);
    }

    render() {
        if ((this.state.windowWidth <= ICONBUTTON_MIN_SCREEN_WIDTH || this.props.behaviour === IconButtonBehavour.SMALL) && this.props.behaviour !== IconButtonBehavour.BIG) {
            return this.renderSmall();
        } else {
            return this.renderBig();
        }
    }

    renderBig() {
        return (
            <button className={classNames('button', this.props.subClass)} onClick={this.props.action}>
                <i className={"icon-" + this.props.icon} /> {this.props.text}
            </button>
        );
    }

    renderSmall() {
        return (
            <Tooltip content={this.props.text} placement={this.props.placement}>
                <button className={classNames('small-button', this.props.subClass)} onClick={this.props.action}>
                    <i className={"icon-" + this.props.icon} />
                </button>
            </Tooltip>
        );
    }
}
