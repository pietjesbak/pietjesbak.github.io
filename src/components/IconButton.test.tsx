import * as enzyme from 'enzyme';
import * as React from 'react';

import IconButton, { IconButtonBehavour } from './IconButton';

jest.mock('react-simple-tooltip', () => {
    return {
        'default': 'mock-tooltip'
    }
});

describe('<IconButton />', () => {
    it('Renders a big button', () => {
        const fn = jest.fn();
        const button = enzyme.shallow(<IconButton icon="globe" text="Home" action={fn} />);

        expect(button).toMatchSnapshot('big button');
        button.find('button').simulate('click');
        expect(fn).toBeCalled();
    });

    it('Renders a small button', () => {
        (window as any).innerWidth = 500;

        const fn = jest.fn();
        const button = enzyme.mount(<IconButton icon="globe" text="Home" action={fn} />);

        expect(button).toMatchSnapshot('small button');
        button.find('button').simulate('click');
        expect(fn).toBeCalled();
    });

    it('Renders a big button regardless of screen size', () => {
        (window as any).innerWidth = 500;

        const fn = jest.fn();
        const button = enzyme.shallow(<IconButton icon="globe" text="Home" action={fn} behaviour={IconButtonBehavour.BIG} />);

        expect(button).toMatchSnapshot('big button');
        button.find('button').simulate('click');
        expect(fn).toBeCalled();
    });

    it('Renders a small button regardless of screen size', () => {
        (window as any).innerWidth = 1920;

        const fn = jest.fn();
        const button = enzyme.shallow(<IconButton icon="globe" text="Home" action={fn} behaviour={IconButtonBehavour.SMALL} />);

        expect(button).toMatchSnapshot('small button');
        button.find('button').simulate('click');
        expect(fn).toBeCalled();
    });
});
