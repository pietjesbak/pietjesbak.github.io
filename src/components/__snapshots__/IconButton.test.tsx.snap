// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> Renders a big button regardless of screen size: big button 1`] = `
<button
  className="button"
  onClick={[MockFunction]}
>
  <i
    className="icon-globe"
  />
   
  Home
</button>
`;

exports[`<IconButton /> Renders a big button: big button 1`] = `
<button
  className="button"
  onClick={[MockFunction]}
>
  <i
    className="icon-globe"
  />
   
  Home
</button>
`;

exports[`<IconButton /> Renders a small button regardless of screen size: small button 1`] = `
<mock-tooltip
  content="Home"
>
  <button
    className="small-button"
    onClick={[MockFunction]}
  >
    <i
      className="icon-globe"
    />
  </button>
</mock-tooltip>
`;

exports[`<IconButton /> Renders a small button: small button 1`] = `
<IconButton
  action={[MockFunction]}
  behaviour={2}
  icon="globe"
  text="Home"
>
  <mock-tooltip
    content="Home"
  >
    <button
      className="small-button"
      onClick={[MockFunction]}
    >
      <i
        className="icon-globe"
      />
    </button>
  </mock-tooltip>
</IconButton>
`;
